kernel/src/path.rs

```diff
@@ -180,22 +181,59 @@ impl<Location: AsUrl> ParsedLogPath<Location> {
 }
 
 impl ParsedLogPath<Url> {
-    /// Create a new ParsedCommitPath<Url> for a new json commit file at the specified version
-    pub(crate) fn new_commit(
-        table_root: &Url,
-        version: Version,
-    ) -> DeltaResult<ParsedLogPath<Url>> {
+    const DELTA_LOG_DIR: &'static str = "_delta_log/";
+
+    /// Helper method to create a path with the given filename generator
+    fn create_path(table_root: &Url, filename: String) -> DeltaResult<Self> {
+        let location = table_root.join(Self::DELTA_LOG_DIR)?.join(&filename)?;
+        Self::try_from(location)?
+            .ok_or_else(|| Error::internal_error("attempted to create invalid path"))
+    }
+
+    /// Create a new ParsedCommitPath<Url> for a new json commit file
+    pub(crate) fn new_commit(table_root: &Url, version: Version) -> DeltaResult<Self> {
         let filename = format!("{:020}.json", version);
-        let location = table_root.join("_delta_log/")?.join(&filename)?;
-        let path = Self::try_from(location)?
-            .ok_or_else(|| Error::internal_error("attempted to create invalid commit path"))?;
+        let path = Self::create_path(table_root, filename)?;
         if !path.is_commit() {
             return Err(Error::internal_error(
                 "ParsedLogPath::new_commit created a non-commit path",
             ));
         }
         Ok(path)
     }
+
+    /// Create a new ParsedCheckpointPath<Url> for a classic parquet checkpoint file
+    #[allow(dead_code)] // TODO: Remove this once we have a use case for it
+    pub(crate) fn new_classic_parquet_checkpoint(
+        table_root: &Url,
+        version: Version,
+    ) -> DeltaResult<Self> {
+        let filename = format!("{:020}.checkpoint.parquet", version);
+        let path = Self::create_path(table_root, filename)?;
+        if !path.is_checkpoint() {
+            return Err(Error::internal_error(
+                "ParsedLogPath::new_classic_parquet_checkpoint created a non-checkpoint path",
+            ));
+        }
+        Ok(path)
+    }
+
+    /// Create a new ParsedCheckpointPath<Url> for a UUID-based parquet checkpoint file
+    #[allow(dead_code)] // TODO: Remove this once we have a use case for it
+    pub(crate) fn new_uuid_parquet_checkpoint(
+        table_root: &Url,
+        version: Version,
+    ) -> DeltaResult<Self> {
+        let uuid = Uuid::new_v4().to_string();
+        let filename = format!("{:020}.checkpoint.{}.parquet", version, uuid);
```

Comment:
I believe `Uuid` is `ToString` and so we can do:
```suggestion
        let filename = format!("{:020}.checkpoint.{}.parquet", version, Uuid::new_v4());
```